src/containers/MyNdla/MyProfile/components/MyPreferences.tsx

```diff
@@ -0,0 +1,142 @@
+import { FeideUserApiType } from '../../../../interfaces';
+import { isStudent } from '../../Folders/util';
+
+type MyPreferencesProps = {
+  user: FeideUserApiType | undefined;
+};
+
+const PreferenceContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: ${spacing.normal};
+  max-width: 700px;
+`;
+
+const DisclaimerContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: ${spacing.small};
+`;
+
+const OptionContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: ${spacing.small};
+`;
+
+const StyledRadioButtonGroup = styled(RadioButtonGroup)`
+  gap: 0px;
+  max-width: 380px;
+  padding: 0;
+  > div {
+    box-sizing: content-box;
+    border-radius: ${misc.borderRadius};
+    border: 1px solid ${colors.brand.greyLight};
+    border-radius: 0px;
+    padding: ${spacing.small} ${spacing.normal};
+    border-color: ${colors.brand.light};
+    &:focus-within,
+    &[data-state='checked'] {
+      outline: 0px;
+      border-color: ${colors.brand.primary};
+      border-radius: 0px;
+      z-index: 1;
+    }
+    &:first-of-type {
+      border-top-left-radius: ${misc.borderRadius};
+      border-top-right-radius: ${misc.borderRadius};
+    }
+    &:not(:first-of-type) {
+      margin-top: -1px;
+    }
+    &:last-of-type {
+      border-bottom-left-radius: ${misc.borderRadius};
+      border-bottom-right-radius: ${misc.borderRadius};
+    }
+    > label {
+      ${fonts.sizes('16px', '24px')}
+      font-weight: ${fonts.weight.semibold};
+    }
+  }
+`;
+
+const MyPreferences = ({ user }: MyPreferencesProps) => {
+  const [_userPreference, setUserPreference] = useState<string>('showName');
+  const { t } = useTranslation();
+
+  return (
+    <PreferenceContainer>
+      <DisclaimerContainer>
+        <Heading
+          element="h2"
+          id="myProfileTitle"
+          margin="none"
+          headingStyle="h2"
+        >
+          {t(
+            `myNdla.myProfile.disclaimerTitle.${
+              isStudent(user) ? 'student' : 'employee'
+            }`,
+          )}
+        </Heading>
+        <Text element="p" textStyle="content-alt" margin="none">
+          {t(
+            `myNdla.myProfile.disclaimerText.${
+              isStudent(user) ? 'student' : 'employee'
+            }`,
+          )}
+        </Text>
+      </DisclaimerContainer>
+      {!isStudent(user) && (
+        <>
+          <OptionContainer>
+            <Heading
+              element="h2"
+              id="myProfileTitle"
+              margin="none"
+              headingStyle="h2"
+            >
+              {t('myNdla.myProfile.preferenceTitle')}
+            </Heading>
+            <Text element="p" textStyle="content-alt" margin="none">
+              {t('myNdla.myProfile.preferenceText')}
+            </Text>
+          </OptionContainer>
+          <StyledRadioButtonGroup
+            options={[
+              {
+                title: t('myNdla.myProfile.radioButtonText.option1'),
```

Comment:
Navna her seier ingenting. Det kunne vore definert som myNdla.myProfile.namePreference.showName og myNdla.myProfile.namePreference.dontShowName

Reply:
prøvde å være litt generisk i tilfelle de endrer seg, men vil du da at jeg skal bytte i frontend-packages?

Reply:
Generisk kan være greit av og til, men her vil det nok ikkje være behov for endringer. Så eg foretrekker at navninga er mest mulig tydelig.